### ISSUE TITLE
ToDo: diffs FF52-FF53

### ISSUE THREAD

**Issue by earthlng** [ghacksuserjs/ghacks-user.js#47]
[**v52.0 vs v53.0**](https://github.com/earthlng/FFprefs-diffs/blob/master/diff-v52.0-vs-v53.0.log.js)

133 diffs ( 74 new, 38 gone, 21 different )

## new in v53.0:

* [x] pref("browser.storageManager.enabled", false); https://github.com/ghacksuserjs/ghacks-user.js/commit/5cf2de570a28eefae10f483b60a7ea1d4405a6ee
* [x] pref("browser.tabs.remote.separateFileUriProcess", false); #82 https://github.com/ghacksuserjs/ghacks-user.js/commit/49e2025bc5f82d2d0844abe213a2ab08d65626d4
* [x] pref("browser.urlbar.decodeURLsOnCopy", false); see [comment](https://github.com/ghacksuserjs/ghacks-user.js/issues/47#issuecomment-287546722) , https://github.com/ghacksuserjs/ghacks-user.js/commit/7496b873b78f9cfb82755f4999f97cddce1c75c7
* [x] pref("dom.IntersectionObserver.enabled", false); https://github.com/ghacksuserjs/ghacks-user.js/commit/8ca3176ab13e011b0b3243b58644c208b33955d2 
* [x] pref("extensions.webextensions.remote", false); #82 https://github.com/ghacksuserjs/ghacks-user.js/commit/49e2025bc5f82d2d0844abe213a2ab08d65626d4
* [x] pref("svg.disabled", false); `2671` https://github.com/ghacksuserjs/ghacks-user.js/commit/551427fccc8450598d80186447fcf10240843a70
   * FUKYEAH™ :+1: but inactive due to [youtube](https://github.com/ghacksuserjs/ghacks-user.js/issues/77) :poop: 
* since defaults are false, added to [investigate/keep an eye on sticky](https://github.com/ghacksuserjs/ghacks-user.js/issues/20)
   * pref("dom.dialog_element.enabled", false); 
* since defaults are false, added to [investigate/keep an eye on sticky](https://github.com/ghacksuserjs/ghacks-user.js/issues/20)
   * pref("privacy.trackingprotection.annotate_channels", false);
   * pref("privacy.trackingprotection.lower_network_priority", false);
* since defaults are false, added to [investigate/keep an eye on sticky](https://github.com/ghacksuserjs/ghacks-user.js/issues/20)
   * pref("security.webauth.webauthn", false);
   * pref("security.webauth.webauthn_enable_softtoken", false);
   * pref("security.webauth.webauthn_enable_usbtoken", false);
* pref("network.http.referer.userControlPolicy", 3); `1606`
* pref("privacy.userContext.longPressBehavior", 0);
   * this will be dealt with in #91 : create containers section (containers are off by default in 53)
* pref("services.blocklist.pinning.enabled", true);
   * leaving enabled for now, added to [investigate/keep an eye on sticky](https://github.com/ghacksuserjs/ghacks-user.js/issues/20)
* pref("plugins.navigator.hidden_ctp_plugin", "");
   * ignoring this. flash needs to die. our user.js blocks flash by default. see earthlng's comments below for more info and/or read [1294341](https://bugzilla.mozilla.org/show_bug.cgi?id=1294341)

## removed, renamed or hidden in v53.0:

See https://github.com/ghacksuserjs/ghacks-user.js/commit/d87bcfde58feb316b6f7d947b35243951cd82c36

* [x] pref("dom.beforeAfterKeyboardEvent.enabled", false); [~~1322736~~](https://bugzilla.mozilla.org/show_bug.cgi?id=1322736 "RESOLVED FIXED in Firefox 53 - Drop before and after keydown, keyup events")
* [x] pref("media.getusermedia.screensharing.allow_on_old_platforms", false); [~~1329562~~](https://bugzilla.mozilla.org/show_bug.cgi?id=1329562 "RESOLVED FIXED in Firefox 53 - Remove WebRTC checks for Windows Vista") [2d7af132a40c](https://hg.mozilla.org/mozilla-central/rev/2d7af132a40c "Bug 1329562 - Part 2: Remove gUM screensharing.allow_on_old_platforms check for XP")
* [x] pref("plugin.scan.Acrobat", "5.0"); [~~1317109~~](https://bugzilla.mozilla.org/show_bug.cgi?id=1317109 "RESOLVED FIXED in Firefox 53 - Remove Acrobat plugin quirk handling")
* [x] pref("plugin.scan.Quicktime", "5.0"); [~~1317110~~](https://bugzilla.mozilla.org/show_bug.cgi?id=1317110 "RESOLVED FIXED in Firefox 53 - Remove QuickTime plugin quirk handling")
* [x] pref("plugin.scan.WindowsMediaPlayer", "7.0"); [~~1317108~~](https://bugzilla.mozilla.org/show_bug.cgi?id=1317108 "RESOLVED FIXED in Firefox 53 - Stop scanning for the Windows Media Player plugin")
* [x] pref("security.tls.unrestricted_rc4_fallback", false); [~~1130670~~](https://bugzilla.mozilla.org/show_bug.cgi?id=1130670 "RESOLVED FIXED in Firefox 53 - Remove dead code that handles RC4 fallback")

## changed in v53.0:
(**_striked ones changed in a beta version but the change didn't land in stable_**)
* [x] pref("webextensions.storage.sync.enabled", true); // prev: false https://github.com/ghacksuserjs/ghacks-user.js/commit/ec5fdfcdaa9f8f5e9853596ed5455f2f6032ef54
* ~~pref("browser.crashReports.unsubmittedCheck.enabled", true); // prev: false~~ `0351`
* ~~pref("browser.migrate.automigrate.enabled", true); // prev: false~~ `3023`
* pref("browser.safebrowsing.provider.google4.gethashURL", ""); // prev: somethingsomething `0410c`
* pref("media.getusermedia.browser.enabled", false); // prev: true `2022`
* pref("network.predictor.enable-prefetch", false); // prev: true `0608`
* pref("security.pki.certificate_transparency.mode", 0); // prev: 1 `already keeping an eye on` (#20)
* ~~pref("security.tls.version.max", 4); // prev: 3~~ `1202`
* pref("webgl.enable-debug-renderer-info", true); // prev: false `2011`

## ignore
==NEW
```js
pref("browser.formautofill.experimental", false);
pref("browser.safebrowsing.temporary.take_v4_completion_result", false);
pref("devtools.command-button-pick.enabled", true);
pref("devtools.debugger.call-stack-visible", false);
pref("devtools.debugger.end-panel-collapsed", false);
pref("devtools.debugger.pending-selected-location", "{}");
pref("devtools.debugger.scopes-visible", false);
pref("devtools.debugger.start-panel-collapsed", false);
pref("devtools.debugger.tabs", "[]");
pref("devtools.gridinspector.showGridLineNumbers", false);
pref("devtools.gridinspector.showInfiniteLines", false);
pref("devtools.inspector.colorWidget.enabled", false);
pref("devtools.screenshot.audio.enabled", true);
pref("devtools.screenshot.clipboard.enabled", false);
pref("devtools.webconsole.filter.css", false);
pref("dom.forms.selectSearch", false);
pref("dom.ipc.processCount.webLargeAllocation", 10);
pref("dom.largeAllocationHeader.enabled", true);
pref("dom.select_popup_in_parent.enabled", false);
pref("dom.storage.testing", false);
pref("extensions.getAddons.themes.browseURL", "https://addons.mozilla.org/%LOCALE%/firefox/themes/?src=firefox");
pref("extensions.webcompat-reporter.enabled", false);
pref("extensions.webcompat-reporter.newIssueEndpoint", "https://webcompat.com/issues/new");
pref("extensions.webextensions.identity.redirectDomain", "extensions.allizom.org");
pref("extensions.webextensions.themes.enabled", false);
pref("identity.fxaccounts.settings.devices.uri", "https://accounts.firefox.com/settings/clients?service=sync&context=fx_desktop_v3");
pref("layers.geometry.basic.enabled", true);
pref("layers.geometry.opengl.enabled", true);
pref("layers.gpu-process.enabled", true);
pref("layout.animation.prerender.absolute-limit-x", 4096);
pref("layout.animation.prerender.absolute-limit-y", 4096);
pref("layout.animation.prerender.partial", false);
pref("layout.animation.prerender.viewport-ratio-limit-x", "1.125");
pref("layout.animation.prerender.viewport-ratio-limit-y", "1.125");
pref("layout.css.display-flow-root.enabled", true);
pref("layout.css.font-variations.enabled", false);
pref("media.cubeb.log_level", "");
pref("media.decoder.recycle.enabled", false);
pref("media.gpu-process-decoder", true);
pref("media.wmf.allow-unsupported-resolutions", false);
pref("network.standard-url.enable-rust", false);
pref("places.frecency.redirectSourceVisitBonus", 25);
pref("plugins.flashBlock.enabled", false);
pref("privacy.history.custom", false);
pref("privacy.permissionPrompts.showCloseButton", false);
pref("privacy.temporary_permission_expire_time_ms", 3600000);
pref("security.mixed_content.hsts_priming_request_timeout", 3000);
pref("services.blocklist.pinning.bucket", "pinning");
pref("services.blocklist.pinning.checked", 0);
pref("services.blocklist.pinning.collection", "pins");
pref("urlclassifier.flashAllowExceptTable", "testexcept-flashallow-simple,except-flashallow-digest256");
pref("urlclassifier.flashAllowTable", "test-flashallow-simple,allow-flashallow-digest256");
pref("urlclassifier.flashExceptTable", "testexcept-flash-simple,except-flash-digest256");
pref("urlclassifier.flashSubDocExceptTable", "testexcept-flashsubdoc-simple,except-flashsubdoc-digest256");
pref("urlclassifier.flashSubDocTable", "test-flashsubdoc-simple,block-flashsubdoc-digest256");
pref("urlclassifier.flashTable", "test-flash-simple,block-flash-digest256");
pref("webgl.max-acceptable-fb-status-invals", 0);
pref("webgl.max-perf-warnings", 0);
```

==REMOVED or HIDDEN
```js
/*** we have none of these ***/
pref("accessibility.ipc_architecture.enabled", true);
pref("browser.preferences.animateFadeIn", false);
pref("browser.tabs.dontfocusfordialogs", true);
pref("browser.uitour.readerViewTrigger", "^https:\\/\\/www\\.mozilla\\.org\\/[^\\/]+\\/firefox\\/reading\\/start");
pref("devtools.apps.forbidden-permissions", "embed-apps");
pref("devtools.netmonitor.statistics", true);
pref("dom.details_element.enabled", true);
pref("dom.mozInputMethod.enabled", false);
pref("dom.mozNetworkStats.enabled", false);
pref("dom.mozPermissionSettings.enabled", false);
pref("dom.mozSettings.allowForceReadOnly", false);
pref("dom.mozSettings.enabled", false);
pref("dom.mozSettings.SettingsDB.debug.enabled", false);
pref("dom.mozSettings.SettingsDB.verbose.enabled", false);
pref("dom.mozSettings.SettingsManager.debug.enabled", false);
pref("dom.mozSettings.SettingsManager.verbose.enabled", false);
pref("dom.mozSettings.SettingsRequestManager.debug.enabled", false);
pref("dom.mozSettings.SettingsRequestManager.verbose.enabled", false);
pref("dom.mozSettings.SettingsService.debug.enabled", false);
pref("dom.mozSettings.SettingsService.verbose.enabled", false);
pref("dom.node.rootNode.enabled", false);
pref("dom.presentation.discovery.legacy.enabled", false);
pref("dom.system_update.debug", false);
pref("dom.system_update.enabled", false);
pref("gecko.handlerService.allowRegisterFromDifferentHost", false);
pref("gfx.prefer-mesa-llvmpipe", false);
pref("image.mozsamplesize.enabled", false);
pref("layout.css.display-contents.enabled", true);
pref("network.standard-url.encode-utf8", true);
pref("network.standard-url.escape-utf8", true);
pref("security.ssl.enable_npn", true);
pref("security.ssl.false_start.require-npn", false);
```

==CHANGED
```js
pref("browser.addon-watch.ignore", "[\"mochikit@mozilla.org\",\"special-powers@mozilla.org\",\"fxdevtools-adapters@mozilla.org\",\"fx-devtools\",\"webcompat-reporter@mozilla.org\"]"); // prev: "[\"mochikit@mozilla.org\",\"special-powers@mozilla.org\",\"fxdevtools-adapters@mozilla.org\",\"fx-devtools\"]"
pref("browser.migrate.chrome.history.limit", 2000); // prev: 0
pref("browser.migrate.chrome.history.maxAgeInDays", 180); // prev: 0
pref("browser.safebrowsing.provider.mozilla.lists", "base-track-digest256,mozstd-trackwhite-digest256,content-track-digest256,mozplugin-block-digest256,mozplugin2-block-digest256,block-flash-digest256,except-flash-digest256,allow-flashallow-digest256,except-flashallow-digest256,block-flashsubdoc-digest256,except-flashsubdoc-digest256"); // prev: "base-track-digest256,mozstd-trackwhite-digest256,content-track-digest256,mozplugin-block-digest256,mozplugin2-block-digest256"
pref("devtools.jsonview.enabled", true); // prev: false
pref("devtools.netmonitor.har.defaultFileName", "Archive %date"); // prev: "Archive %y-%m-%d %H-%M-%S"
pref("devtools.toolbox.toolbarSpec", '["splitconsole", "paintflashing toggle","scratchpad","resize toggle","screenshot --fullpage --file", "rulers", "measure"]'); // prev: '["splitconsole", "paintflashing toggle","scratchpad","resize toggle","screenshot --fullpage", "rulers", "measure"]'
pref("lightweightThemes.recommendedThemes", /*** big long string ***/); // prev: /*** big long string ***/
pref("media.decoder-doctor.notifications-allowed", /*** big long string ***/); // prev: /*** big long string ***/
pref("media.dormant-on-pause-timeout-ms", 5000); // prev: -1
pref("media.navigator.audio.full_duplex", true); // prev: false
pref("places.frecency.permRedirectVisitBonus", 50); // prev: 0
pref("places.frecency.tempRedirectVisitBonus", 40); // prev: 0
pref("services.sync.engine.tabs.filteredUrls", "^(about:.*|resource:.*|chrome:.*|wyciwyg:.*|file:.*|blob:.*)$"); // prev: "^(about:.*|chrome://weave/.*|wyciwyg:.*|file:.*|blob:.*)$"
pref("urlclassifier.disallow_completions", /*** big long string ***/); // prev: /*** big long string ***/
```

**Comment by earthlng** [comment]
## Passive TrackingProtection

[Lower priority of HTTP requests for resources on the Tracking Protection list](https://bugzilla.mozilla.org/show_bug.cgi?id=1141814)	-> RESOLVED FIXED in Firefox 53

> We could add a new pref here like "privacy.trackingprotection.annotate_channels" or something like that. If it's off, then we won't update the list or annotate the channels and so none of the perf features will do anything.

[Part 1: Enable to update TP list if TP is disabled. r=francois](https://hg.mozilla.org/mozilla-central/rev/fc772f9c5fb4)

[Part 2: Lower the priority of channel loading tracking resource. r=honzab](https://hg.mozilla.org/mozilla-central/rev/fb18b507a912)

```js
// Annotate channels based on the tracking protection list in all modes
pref("privacy.trackingprotection.annotate_channels",  false);
```

---

[Add a passive (detection only) mode for Tracking Protection](https://bugzilla.mozilla.org/show_bug.cgi?id=1170190)	-> RESOLVED FIXED in Firefox 53

[Part 1: Split out a new pref from privacy.trackingprotection.annotate_channels to explicitly control whether the channel priority is adjusted; r=mayhemer](https://hg.mozilla.org/mozilla-central/rev/d2c65588f41f)

```js
// Lower the priority of network loads for resources on the tracking protection list.
// Note that this requires the privacy.trackingprotection.annotate_channels pref to be on in order to have any effect.
pref("privacy.trackingprotection.lower_network_priority",  false);
```

**Comment by earthlng** [comment]
[Remove Acrobat plugin quirk handling](https://bugzilla.mozilla.org/show_bug.cgi?id=1317109) ( RESOLVED FIXED in Firefox 53 )

https://hg.mozilla.org/mozilla-central/rev/c8f05f715087

```
-// Locate plugins by scanning the Adobe Acrobat installation directory with a minimum version
-pref("plugin.scan.Acrobat", "5.0");
```

[Remove QuickTime plugin quirk handling	](https://bugzilla.mozilla.org/show_bug.cgi?id=1317110) ( RESOLVED FIXED in Firefox 53 )

https://hg.mozilla.org/mozilla-central/rev/6abf12a9d408

```
-// Locate plugins by scanning the Quicktime installation directory with a minimum version
-pref("plugin.scan.Quicktime", "5.0");
```

[Stop scanning for the Windows Media Player plugin](https://bugzilla.mozilla.org/show_bug.cgi?id=1317108) ( RESOLVED FIXED in Firefox 53 )

https://hg.mozilla.org/mozilla-central/rev/510f51dd6822
```

-// Locate and scan the Window Media Player installation directory for plugins with a minimum version
-pref("plugin.scan.WindowsMediaPlayer", "7.0");
```

**Comment by earthlng** [comment]
### services.blocklist.pinning.*

[Create a services client for augmenting the PKP preload list between releases](https://bugzilla.mozilla.org/show_bug.cgi?id=1306470)

> The services blocklist client provides a mechanism we can use to get public key pin preloads to the browser between releases.

https://bugzilla.mozilla.org/show_bug.cgi?id=1306470#c12 :

> It's a little awkward - we're actively reaching into people's profiles and removing a security state the website set for them. So it moves this service from 'Could be used to DOS people by preloading them with invalid pins' to 'Could be used to actively attack them'. So we should be cognizant of that fact and consider how we secure the update mechanism.

```js
pref("services.blocklist.pinning.enabled", true);
pref("services.blocklist.pinning.bucket", "pinning");
pref("services.blocklist.pinning.collection", "pins");
pref("services.blocklist.pinning.checked", 0);
```

**Comment by Roman-Nopantski** [comment]
I was wondering when all the plugin stuff would get dumped - can't wait until the plugins section only has flash in it.

**Comment by earthlng** [comment]
> can't wait until the plugins section only has flash in it.

yeah, because Flash is the best one anyway, amirite? ;)

**Comment by Roman-Nopantski** [comment]
I want that to die too .. but that is in a different time frame. The rest of all that crap should be gone earlier

**Comment by earthlng** [comment]
```js
/******************************************************************************************
 * diffs between FF prefs v53.0b1 and v53.0b2
 ******************************************************************************************/
// removed, renamed or hidden in v53.0b2:
pref("browser.uitour.readerViewTrigger", "^https:\\/\\/www\\.mozilla\\.org\\/[^\\/]+\\/firefox\\/reading\\/start");

// changed in v53.0b2:
pref("browser.migrate.chrome.history.limit", 2000); // prev: 0
pref("browser.migrate.chrome.history.maxAgeInDays", 180); // prev: 0
```

**Comment by earthlng** [comment]
We need to change/adjust the comment for `browser.migrate.automigrate.enabled`
```
/* 3023: disable automigrate, current default is false but may change (FF49+)
 * need more info, but lock down for now ***/
```
for example
```
/* 3023: disable automigrate (FF49+)
 * default is false in FF49+ and true in FF53+
 * need more info, but lock down for now ***/
```

**Comment by Thorin-Oakenpants** [comment]
https://developer.mozilla.org/en-US/Firefox/Experimental_features  scroll down to JavaScript>Shared Memory objects and you will find the experimental feature `javascript.options.shared_memory` which was in the items to check from ghacks days, which is now ticked off due to it being here, and which in FF53 changes from false to true. No idea what it does, I haven't investigate it yet.

**Comment by earthlng** [comment]
`javascript.options.shared_memory` that link mentions 2 new "objects", [SharedArrayBuffer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer) + [Atomics](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Atomics)

from those pages:
> The Atomics object provides atomic operations as static methods. **They are used with SharedArrayBuffer objects**.

> APIs accepting SharedArrayBuffer objects:
> * WebGLRenderingContext.bufferData()
> * WebGLRenderingContext.bufferSubData()
> * WebGL2RenderingContext.getBufferSubData()

it seems to be only used by WebGL and we can safely ignore this pref IMO.

**Comment by earthlng** [comment]
`security.tls.version.max` - if mozilla enables this in FF53 then we should also activate that pref to enforce it

**Comment by earthlng** [comment]
`security.webauth.webauthn*` - I have enforced these to **false** in my user.js because I have absolutely no need for this anytime soon. https://www.w3.org/TR/webauthn/#use-cases
But atm they are **false** by default so we can either ignore them for now or enforce them to false, perhaps commented out for people who want to use it. But since it's still false by default maybe it's not quite ready yet anyway. Pants, you decide

**Comment by earthlng** [comment]
Why remove the discussion and help wanted labels?

**Comment by Thorin-Oakenpants** [comment]
i'm anal: all diffs use enhancement & task, all redos use discussion & enhance. Any issue raised that results in a change = enhanced. I want to be able to use the labels - so clicking enhanced loads all the changes we made. Clicking discussion loads all the revamps. That kind of thing.

The issues list was getting too colorful and epileptic :) OK, I made that one up. Help wanted is a default label, I don't even like it. EVERY issue is a help wanted - comments welcome. Besides, until we have some more members, between you and me (and chip ins from crssi and Atavic), we can easily narrow it down to the 10-20 prefs and investigate. The deprecated is all done bar checking once stable lands. The changed is almost all done as well. Help/comments/questions/issues are all welcome by default - don't think we need a label for it - besides, I hate that it makes us "look" inadequate or knowledgeable - like we're begging for assistance.

**Comment by earthlng** [comment]
changes in beta9 - I'll add them to the first comment

```js
/******************************************************************************************
 * diffs between FF prefs v53.0b1 and v53.0b9
 ******************************************************************************************/
// new in v53.0b9:
pref("plugins.navigator.hidden_ctp_plugin", "");

// removed, renamed or hidden in v53.0b9:
pref("browser.migrate.automigrate.undo-survey", "https://qsurvey.mozilla.com/s3/Firefox-52-Import-Undo?prevBrowser=%IMPORTEDBROWSER%");
pref("browser.migrate.automigrate.undo-survey-locales", "en-US");
pref("browser.uitour.readerViewTrigger", "^https:\\/\\/www\\.mozilla\\.org\\/[^\\/]+\\/firefox\\/reading\\/start");

// changed in v53.0b9:
pref("browser.crashReports.unsubmittedCheck.enabled", false); // prev: true
pref("browser.migrate.automigrate.enabled", false); // prev: true
pref("browser.migrate.chrome.history.limit", 2000); // prev: 0
pref("browser.migrate.chrome.history.maxAgeInDays", 180); // prev: 0
pref("browser.safebrowsing.provider.mozilla.lists", "base-track-digest256,mozstd-trackwhite-digest256,content-track-digest256,mozplugin-block-digest256,mozplugin2-block-digest256,block-flash-digest256,except-flash-digest256,allow-flashallow-digest256,except-flashallow-digest256,block-flashsubdoc-digest256,except-flashsubdoc-digest256"); // prev: "base-track-digest256,mozstd-trackwhite-digest256,content-track-digest256,mozplugin-block-digest256,mozplugin2-block-digest256"
pref("network.predictor.enable-prefetch", false); // prev: true
pref("urlclassifier.disallow_completions", "test-malware-simple,test-phish-simple,test-unwanted-simple,test-track-simple,test-trackwhite-simple,test-block-simple,test-flashallow-simple,testexcept-flashallow-simple,test-flash-simple,testexcept-flash-simple,test-flashsubdoc-simple,testexcept-flashsubdoc-simple,goog-downloadwhite-digest256,base-track-digest256,mozstd-trackwhite-digest256,content-track-digest256,mozplugin-block-digest256,mozplugin2-block-digest256,block-flash-digest256,except-flash-digest256,allow-flashallow-digest256,except-flashallow-digest256,block-flashsubdoc-digest256,except-flashsubdoc-digest256"); // prev: "test-malware-simple,test-phish-simple,test-unwanted-simple,test-track-simple,test-trackwhite-simple,test-block-simple,test-flashallow-simple,testexcept-flashallow-simple,test-flash-simple,testexcept-flash-simple,test-flashsubdoc-simple,testexcept-flashsubdoc-simple,goog-downloadwhite-digest256,base-track-digest256,mozstd-trackwhite-digest256,content-track-digest256,mozplugin-block-digest256,mozplugin2-block-digest256"
pref("urlclassifier.flashAllowExceptTable", "testexcept-flashallow-simple,except-flashallow-digest256"); // prev: "testexcept-flashallow-simple"
pref("urlclassifier.flashAllowTable", "test-flashallow-simple,allow-flashallow-digest256"); // prev: "test-flashallow-simple"
pref("urlclassifier.flashExceptTable", "testexcept-flash-simple,except-flash-digest256"); // prev: "testexcept-flash-simple"
pref("urlclassifier.flashSubDocExceptTable", "testexcept-flashsubdoc-simple,except-flashsubdoc-digest256"); // prev: "testexcept-flashsubdoc-simple"
pref("urlclassifier.flashSubDocTable", "test-flashsubdoc-simple,block-flashsubdoc-digest256"); // prev: "test-flashsubdoc-simple"
pref("urlclassifier.flashTable", "test-flash-simple,block-flash-digest256"); // prev: "test-flash-simple"

```

**Comment by Thorin-Oakenpants** [comment]
While you're there @earthlng : [here](https://github.com/ghacksuserjs/ghacks-user.js/wiki/Appendix-D:-References) ... i'm busy going thru my privacy/web/tech speed dial tabs and whatnot building a list

**Comment by earthlng** [comment]
placeholder (?) - yeah looks great so far :)

**Comment by earthlng** [comment]
what do you mean "while you're there"? I'm always here :)

**Comment by earthlng** [comment]
https://www.mozilla.org/en-US/security/known-vulnerabilities/firefox/
https://blog.mozilla.org/security/category/security/
https://www.mozilla.org/en-US/firefox/releases/
https://developer.mozilla.org/en-US/Firefox/Releases
https://www.fxsitecompat.com/en-CA/
https://www.fxsitecompat.com/en-CA/docs/

**Comment by earthlng** [comment]
* I moved the following to ignore because the main pref seems to be `services.blocklist.pinning.enabled`
pref("services.blocklist.pinning.bucket", "pinning");
pref("services.blocklist.pinning.collection", "pins");
* `svg.disabled`
I hate to say it, especially after waiting for this for so long, but I think we should comment this out because it breaks youtube player controls.
* `browser.storageManager.enabled`
from [here](https://blog.mozilla.org/l10n/2017/03/07/firefox-l10n-report-aurora-54/)

> In the last couple of cycles, some strings landed in pref for managing Site Data. To see this section in Preferences (at the bottom of Advanced -> Network), you need to enable (set to “true”) both these keys in about:config
> 
>     browser.storageManager.enabled
>     dom.storageManager.enabled
> 
> **Functionality is still hard to test, since there are no websites using this feature available for testing.**

=> let's add this to 2706

* `plugins.navigator.hidden_ctp_plugin`
[1294341](https://bugzilla.mozilla.org/show_bug.cgi?id=1294341) - something to do with Click2Play and Flash not being detected
From [here](https://hg.mozilla.org/integration/mozilla-inbound/rev/be81e6841555)

> // **This only supports one hidden ctp plugin**, edit nsPluginArray.cpp if adding a second
> pref("plugins.navigator.hidden_ctp_plugin", "Shockwave Flash");

=> the default empty string seems fine since we strongly recommend to not use Flash. IMO we can ignore this pref.

* `devtools.jsonview.enabled`
Why do you think we need to deal with this one, Pants? seems good to me and we usually ignore `devtools.*` prefs anyway.

**Comment by earthlng** [comment]
`dom.IntersectionObserver.enabled`

Holy fucking shit!!!! This one deserves a special comment!! Started reading about it [here](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API) and it sounded fine to slightly fishy. But then I glanced over [the ticket](https://bugzilla.mozilla.org/show_bug.cgi?id=1243846) and holy fucking shit I can't believe I just read that!

> We're adding a new API and it **will help developers move ad viewability checks** from Flash to JavaScript.

> We're talking to **some ad network partners** about obtaining any tests they might have.

They want to enable it by default soon in [this ticket](https://bugzilla.mozilla.org/show_bug.cgi?id=1321865), but they still have some problems. some gems:

> Given the history of this new API -- it's been the top cause of crashes in Nightly on three different occasions now

> By my count, this is now the 4th time this has landed and been backed out for stability issues

I can't fucking believe that they wasted even a single fucking minute to help Ad networks and crashed Nightly in the process - for A.FUCKING.AD.NETWORK!!! Holy shit, this shattered my believe and trust in mozilla more than anything I have seen so far. WTF are they doing???!!!

Good catch there Pants! I definitely would have missed it if you didn't keep in the to investigate part above. Do I need to say what I think we should do with this pref? ;)

**Comment by crssi** [comment]
No comment. I have donated to FF a few times. I am in a shame for my actions. :(
FF is constantly working on "shady" thingies and also ignoring some real problems about privacy.
More and more I am convinced about smelly background and all this is in purpose. :(

**Comment by earthlng** [comment]
Ok, maybe I over-reacted a bit and to be fair Chrome and IE also support that API so it's not like mozilla came up with the idea. It was also already possible to do this, albeit maybe not as easy, reliable or pixel-accurate:

> The Intersection Observer API addresses the above issues by giving developers a new method to asynchronously query the position of an element with respect to other elements or the global viewport. The asynchronous delivery **eliminates the need for costly DOM and style queries, continuous polling, and use of custom plugins**. By removing the need for these methods it allows applications to **significantly reduce their CPU, GPU and energy costs**.

Don't you see? They do it for our benefits! And for the environment ;)

[Happy selling!](https://www.iab.com/viewability-has-arrived-what-you-need-to-know-to-see-through-this-sea-change/) (linked from the spec) :(

**Comment by earthlng** [comment]
> I have donated to FF a few times
Yeah, donating to a multi-million dollar company is probably not the best idea.
Hell, I don't even buy T-shirts with brands on them but that's just me. If anything they would have to pay ME to wear those shirts. But I prefer not to be a walking Ad campaign.

**Comment by zymase** [comment]
@earthlng , you write 

> [...]to be fair Chrome and IE also implemented that API[...]

Maybe but Chrome and IE don't show off with their commitment to users privacy as Mozilla does. 
It appears to those who who believed that Mozilla was apart of the majors when it comes to users' privacy that this is untrue.
My feeling is that the Mozilla mentality is now a blend of complex of inferiority and determination to compare (and beat?) the big names of browsers whatever it may cost. I had an English teacher who told us the word was _upstart crow_. I mean, that's just the way I feel it. I may be wrong. I'd desperately prefer to be wrong.

**Comment by Thorin-Oakenpants** [comment]
> We should start creating Pull Requests for the NEW prefs so we only need to merge them whenever your plan calls for it. I'm still a bit confused about when exactly you want to make those changes

I hate PRs. In a few days we can really get down to deciding what needs to be added and what can be discarded. On April 18 we can archive a 52, move current to 53, easily test and move all deprecated items and start on 53 changes. That's a week away. No hurry :) PS: seems like u might be right about SVG (I dont even use YT) and it may be best as a hardened only version. SVG is pretty damn popular

**Comment by earthlng** [comment]
> I hate PRs. In a few days we can really get down to deciding what needs to be added and what can be discarded

I think that's a bit stupid. We may not always have time to do everything within a few days before and after a new FF is released. With PR's we can prepare, discuss and finalize new items whenever we have some free time during the 6 weeks between releases. When the time comes all we have to do is merge the ones we want and discard the rest.

**Comment by earthlng** [comment]
Why do you hate PR's anyway? Because it creates a lot of commits potentially? It should be possible to prevent that by instead of merging a finalized PR, copy the final code changes, delete the PR and either create a new PR for merging or directly change the master. That way we can have dozens of commits to a PR and only one commit that will end up showing in the commit list. idk if it works that way but we will never know if we don't try

**Comment by Thorin-Oakenpants** [comment]
Already archived 52. There's nothing to say we can't move to 53 and start commits. I'm just "torn" over the next code name and lyric. I've got 9 here to pick from - take ya pick

- Call Me Pants, Maybe
   * Your stare was holding, ripped JEANS, skin was showin'
- All About Those Pants
   * Because you know I'm all about those pants, 'bout those pants, no treble
- Pantslide
   * I took my pants, took em down, I climbed a mountain and I turned around
- Dancing with My Pants
   * If I had the chance, I'd ask the world to dance, and I'll be dancing with my pants
- Gimme Pants Jo'anna
   * Gimme pants, Jo'anna, pants, Jo'anna, pants before the morning come
- I'm Gonna Be (500 Pants)
   * When I wake up, well, I know I'm gonna be, I'm gonna be the pants who wake up next to you
- Achy Breaky Pants
   * But don't tell my pants, my achy breaky pants, I just don't think it'd understand
- Dancing Pants
   * Ooh-ooh, see that girl, watch that scene, dig in the dancing pants
- Pantsthumping
   * I get pulled down, but I get up again, you're never gonna keep me down

I'm partial to the pantsthumping myself

**Comment by earthlng** [comment]
> disable WebExtension sync
> 
>     [1] https://wiki.mozilla.org/WebExtensions/chrome.storage.sync [NEED a better link?]

link is fine. https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/storage/sync is also okay but is also linked on [1]

**Comment by earthlng** [comment]
```js
/* 2426: disable Intersection Observer API (FF53+)
 * [1] https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API
 * [2] https://wicg.github.io/IntersectionObserver/
 * [3] https://bugzilla.mozilla.org/show_bug.cgi?id=1243846 ***/
user_pref("dom.IntersectionObserver.enabled", false);
```
or
```
/* 2426: disable Intersection Observer API (FF53+)
 * [1] https://github.com/ghacksuserjs/ghacks-user.js/issues/47#issuecomment-293303172 ***/
user_pref("dom.IntersectionObserver.enabled", false);
```

Maybe add a note about this allowing for pixel-accurate information about what was actually displayed
and being mostly used by Ad Networks for Ad viewability checks.
But since that's already all quoted in my comment, I think we can just link to my comment and the item is short and sweet.

---
```js
/* 3027: decode URLs on copy from the URL bar (FF53+)
 * [1] https://bugzilla.mozilla.org/show_bug.cgi?id=1320061 ***/
user_pref("browser.urlbar.decodeURLsOnCopy", true);
```
---
```js
/* 0403: disable augmenting the PKP preload list between releases (FF53+)
 * [1] https://bugzilla.mozilla.org/show_bug.cgi?id=1306470#c12 ***/
   // user_pref("services.blocklist.pinning.enabled", false);
```
---
```js
/* 0422: disable passive TrackingProtection (FF53+)
 * [1] https://github.com/ghacksuserjs/ghacks-user.js/issues/47#issuecomment-285638216 ***/
user_pref("privacy.trackingprotection.annotate_channels", false);
user_pref("privacy.trackingprotection.lower_network_priority", false);
```

-> maybe add a note that this requires some URL prefs (0410d ??) are left intact ?

**Comment by earthlng** [comment]
changes between FF53 beta9 and FF53.0 stable

### not changed anymore in stable:
* pref("devtools.devedition.promo.enabled", true); // prev: false
=> still false in FF53.0 [mozilla-release](https://dxr.mozilla.org/mozilla-release/source/devtools/client/preferences/devtools.js#14)

* pref("javascript.options.shared_memory", true); // prev: false
=> still false in FF53.0 [mozilla-release](https://dxr.mozilla.org/mozilla-release/source/modules/libpref/init/all.js#1309)

* pref("security.tls.version.max", 4); // prev: 3
=> still 3 in FF53.0 [mozilla-release](https://dxr.mozilla.org/mozilla-release/source/netwerk/base/security-prefs.js#6)


### changed in stable:
* pref("security.pki.certificate_transparency.mode", 0); // prev: 1
=> https://dxr.mozilla.org/mozilla-release/source/netwerk/base/security-prefs.js#91

**Comment by earthlng** [comment]
Should we start adding bugzilla tickets for deprecated prefs?

Pyllyukko wants to have them for them to consider removing prefs from the user.js:
https://github.com/pyllyukko/user.js/issues/33#issuecomment-134719572

I already have a few in my own user.js for some of the recently removed/deprecated prefs if you're interested

This is the one for [security.tls.unrestricted_rc4_fallback](https://bugzilla.mozilla.org/show_bug.cgi?id=1130670) and the [changeset](https://hg.mozilla.org/mozilla-central/rev/1f7832017dbb)

And I already listed the ones for the old plugin scan prefs [here](https://github.com/ghacksuserjs/ghacks-user.js/issues/47#issuecomment-285639826)

**Comment by Thorin-Oakenpants** [comment]
If you want to find them all and add them all in, go for it. I think it's a bit overkill, seeing as they're dead prefs and we've already confirmed they have been removed (as per the section description). But then again, proof is proof for all the disbelievers :)

**Comment by earthlng** [comment]
Yeah, you're right, we don't need them in the user.js - it only bloats the filesize.
I've added some links to the first comment. There are usually not too many prefs that get removed (4 last time, 6 this time) so it's not tooo much work to add them in the _ToDo diffs_ issues from now on [if I feel like it ;) ]

**Comment by earthlng** [comment]
`privacy.userContext.longPressBehavior`

This only works when Containers are enabled: https://hg.mozilla.org/mozilla-central/rev/f248d089469d#l2.72

// 0 disables long press, 1 when clicked, the menu is shown, 2 the menu is shown after X milliseconds.

**Comment by Thorin-Oakenpants** [comment]
soz guys and gals ... I have been away earning some income - am back now. We'll wrap up the 53 changes in the next few days and do an alpha.

**Comment by earthlng** [comment]
`services.blocklist.pinning.enabled` doesn't fully disable the feature. It still sends out update requests to the following urls (as seen by uMatrix, so last url in the list was requested first) but `services.blocklist.pinning.checked` remained `0` ie didn't get updated.

```
https://content-signature.cdn.mozilla.net/chains/pinning-preload.content-signature.mozilla.org-20170510.prod.chain
https://firefox.settings.services.mozilla.com/v1/buckets/pinning/collections/pins
https://firefox.settings.services.mozilla.com/v1/
https://firefox.settings.services.mozilla.com/v1/buckets/pinning/collections/pins/records?_sort=id
https://firefox.settings.services.mozilla.com/v1/
https://content-signature.cdn.mozilla.net/chains/pinning-preload.content-signature.mozilla.org-20170510.prod.chain
https://firefox.settings.services.mozilla.com/v1/buckets/pinning/collections/pins
https://firefox.settings.services.mozilla.com/v1/
https://firefox.settings.services.mozilla.com/v1/buckets/pinning/collections/pins/records?_sort=-last_modified
https://firefox.settings.services.mozilla.com/v1/
```

curiously it sent the requests twice, once with `records?_sort=-last_modified` and then with `records?_sort=id`. If we want to disable the pinning list update between releases I think we need to also clear `services.blocklist.pinning.collection`
Not unexpected there's no data at the moment and I personally will disable the feature because I can wait the 6 weeks between releases for the preload-list updates.

**Comment by Thorin-Oakenpants** [comment]
So what are you saying? Come on man, spit it out. :)

Q1 - who's blocklists are these? kinto?
Q2 - If we are blocking the blocklist updates (are we blocking them), then how to kill the whole thing
Q3 - I usually manually update FF portable with a new PAF. Does update via applilcation interface also update all the blocklists etc?

**Comment by earthlng** [comment]
Seriously? Scroll up slowly all the way to the top and notice all the comments by earthlng - and now you really want me to spoon-feed you even more? Where's the _collaboration_ here?

A1 - mozilla - kinto? yes
A2 - we are not. How come I know your baby better than you dude? :)
A3 - <installdir>/browser/blocklist.xml got updated, but idk if the kinto lists are shipped with the setup, maybe in omni.ja, idk
A4 - yes, blocklist (old) + addons + certs + now maybe pinning, idk
A5 - nothing is strictly necessary

**Comment by Thorin-Oakenpants** [comment]
> How come I know your baby better than you dude

I wanted confirmation that it was kinto - in which case the answer is yes and no - we block some in the default user.js. And the question was how to kill the whole thing, but that's kinda moot for me right now - see next sentences.

I'm of the opinion that we should leave this pinning alone
- does killing the pinning actually kill the list updates?
- if so, this should be left alone - or at the very least confirm all blocklists are updated via all app update methods

**Comment by earthlng** [comment]
The feature of getting pinning updates between FF releases is totally independent of the remaining kinto updates. To kill the whole thing you could just clear the URL pref and all the collection prefs and services.blocklist.update_enabled or set the interval to a gazillion years or something. And then there's also extensions.blocklist.enabled. Currently (and I suspect it will stay that way) the pinning data update list is empty. But it does 10 requests every 24 hours, basically for no reason because the data is empty.
6 weeks between releases is really not that long and idk what would warrant an update in between.
They wanted to have a way to update it and kinto makes this very easy but I think it's very possible that this will never be used. Maybe they'll eventually get rid of the hardcoded preload list and use the kinto list instead but they never said anything to that effect in the ticket.
Since we already have the other kinto prefs in the user.js I thought we should also include these new ones but at this point I really don't care anymore. I've made the decision for my own user.js and whatever you want to with it for this user.js is up to you.

**Comment by Thorin-Oakenpants** [comment]
All done (might still add some items during stable life-cycle from "items to investigate/keep an eye on" - but they are at FF53 defaults at what we would want anyway). There's also an enhancement issue in for adding a containers section - but again, all these prefs are off in FF53, so no need to rush them out in alpha, we can add them in our own sweet time.

Closing this and releasing an alpha for earthlng to do a changelog
